website_sale_require_legal/controllers/main.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingeniería, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.http import request
+from openerp.sql_db import TestCursor
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class RequireLegalTermsToCheckout(website_sale):
+    def _get_mandatory_billing_fields(self, *args, **kwargs):
+        """Require accepting legal terms for unauthenticated users."""
+        result = (super(RequireLegalTermsToCheckout, self).
+                  _get_mandatory_billing_fields(*args, **kwargs))
+
+        # Avoid PhantomJS errors in test mode
+        if not isinstance(request.env.cr, TestCursor):
+            if request.env.user == request.website.user_id:
```

Comment:
@rafaelbn @carlos-incaser I can reproduce the bug in runbot, but not in real machine. Please test in local and provide feedback.

This line seems to evaluate to True (public user) when a user is logged in in the runbot, while it should be False (logged user). That seems the most probable cause of the bug, but no idea why. Any help? CC @pedrobaeza.